drivers/usb/device/usb_dc_it82xx2.c

```diff
@@ -0,0 +1,1687 @@
+
+static const struct usb_it8xxx2_wuc usb_wuc0[IT8XXX2_DT_INST_WUCCTRL_LEN(0)] =
+		IT8XXX2_DT_WUC_ITEMS_LIST(0);
+
+PINCTRL_DT_INST_DEFINE(0);
+
+static const struct usb_it82xx2_config ucfg0 = {
+	.pcfg = PINCTRL_DT_INST_DEV_CONFIG_GET(0),
+	.wuc_list = usb_wuc0
+};
+
+struct usb_device_ep_data {
+	/* Endpoint IN callback function */
+	usb_dc_ep_callback cb_in;
+
+	/* Endpoint OUT callback function */
+	usb_dc_ep_callback cb_out;
+
+	/* Endpoint status */
+	enum ep_status ep_status;
+
+	/* need to be modify after zephyr 2.3 */
+	enum usb_dc_ep_transfer_type ep_type;
+
+	uint16_t remaining; /* remaining bytes */
+
+	/* Max ep pkt size */
+	uint16_t mps;
+};
+
+struct usb_device_data {
+	struct usb_device_ep_data ep_data[MAX_NUM_ENDPOINTS];
+	enum usb_device_state state;
+	enum setup_stage st_state; /* Setup State */
+
+	/* EP0 status */
+	enum trans_type last_token;
+
+	/* EP0 status */
+	enum trans_type now_token;
+
+	uint8_t attached;
+	uint8_t addr;
+	bool no_data_ctrl;
+	bool suspended;
+
+	/* Check the EP interrupt state for (ep > 0) */
+	bool ep_ready[3];
+};
+
+/*
+ * Mapped to the bit definitions in the EPN_EXTEND_CONTROL1 Register
+ * (D6h to DDh) for configuring the FIFO direction and for enabling/disabling
+ * the endpoints.
+ */
+static uint8_t ext_ep_bit_shift[12] = {0, 0, 0, 1, 1, 1, 2, 2, 2, 3, 3, 3};
+
+/*
+ * The ep_fifo_res[ep_idx % FIFO_NUM] where the FIFO_NUM is 3 represents the
+ * EP mapping because when (ep_idx % FIFO_NUM) is 3, it actually means the EP0.
+ */
+static const uint8_t ep_fifo_res[3] = {3, 1, 2};
+
+/*
+ * Mapping the enum extend_ep_ctrl code to their corresponding bit in
+ * the EP45/67/89/1011/1213/1415 Extended Control Registers.
+ */
+static const uint8_t ext_ctrl_tbl[7] = {
+	EPN0_ISO_ENABLE,
+	EPN0_ISO_ENABLE,
+	EPN0_SEND_STALL,
+	EPN0_SEND_STALL,
+	EPN0_SEND_STALL,
+	EPN0_OUTDATA_SEQ,
+	EPN0_OUTDATA_SEQ,
+};
+
+/*
+ * Indexing of the following control codes:
+ * EXT_EP_DIR_IN, EXT_EP_DIR_OUT, EXT_EP_ENABLE, EXT_EP_DISABLE
+ */
+static const uint8_t epn_ext_ctrl_tbl[4] = {1, 0, 1, 0};
+
+static usb_dc_status_callback usb_status_cb;
+static struct usb_device_data udata0 = {
+	.attached = 0,
+	.st_state = 0,
+	.no_data_ctrl = false,
+	.suspended = false,
+};
+
+/*
+ * Abbreviations of the USB Slave Control Registers
+ */
+struct usb_it82xx2_regs *const usb_regs =
+	(struct usb_it82xx2_regs *)IT82XX2_USB_BASE;
+
+struct it82xx2_usb_ep_regs *ep_regs =
+	usb_regs->usb_ep_regs;
+
+struct it82xx2_usb_ep_fifo_regs *ff_regs =
+	usb_regs->fifo_regs;
+
+volatile uint8_t *epn0n1_ext_ctrl =
+	usb_regs->fifo_regs[EP_EXT_REGS_9X].ext_4_15.EPN0N1_EXT_CTRL;
+
+volatile uint8_t *ep_fifo_ctrl =
+	usb_regs->fifo_regs[EP_EXT_REGS_BX].fifo_ctrl.EP_FIFO_CTRL;
+
+struct epn_ext_ctrl_regs *epn_ext_ctrl =
+	usb_regs->fifo_regs[EP_EXT_REGS_DX].ext_0_3.EPN_EXT_CTRL;
+
+
+/*
+ * Function usb_it8xxx2_enable_wu90_irq():
+ *
+ * WU90 (USB D+) Enable/Disable
+ */
+void usb_it8xxx2_enable_wu90_irq(const struct device *dev, bool enable)
```

Comment:
Missing static, please check everywhere.

Reply:
All the functions which should be static have been modified accordingly. Thanks you for your reminding, sir.